### PR TITLE
test: populate exitPool tests

### PR DESCRIPTION
**Pull request by 0xAustrian**

### CODE REVIEW

File: test/unit/BPool.t.sol [hunk 1]
```diff
@@ -702,32 +702,134 @@ contract BPool_Unit_ExitPool is BasePoolTest {
...
   }
 
-  function test_Revert_NotFinalized() private view {}
+  function test_Revert_NotFinalized(ExitPool_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    _setFinalize(false);
 
-  function test_Revert_MathApprox() private view {}
+    vm.expectRevert('ERR_NOT_FINALIZED');
+    bPool.exitPool(_fuzz.poolAmountIn, _zeroAmountsArray());
+  }
```
Comment (wei3erHase) [inline test/unit/BPool.t.sol:713]: we can ignore the happyPath here if i'm not mistaken, just call the pol with anything when is not finalized and it'll revert

File: test/unit/BPool.t.sol [hunk 2]
```diff
@@ -702,32 +702,134 @@ contract BPool_Unit_ExitPool is BasePoolTest {
...
+    bPool.exitPool(_fuzz.poolAmountIn, _zeroAmountsArray());
 
-  function test_Revert_TokenArrayMathApprox() private view {}
+    assertEq(bPool.balanceOf(address(this)), _fuzz.initPoolSupply - _fuzz.poolAmountIn);
+  }
 
-  function test_Revert_TokenArrayLimitOut() private view {}
+  function test_Push_PoolShare(ExitPool_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    uint256 _exitFee = bmul(_fuzz.poolAmountIn, EXIT_FEE);
+    uint256 _balanceBefore = bPool.balanceOf(bPool.call__factory());
```
Comment (wei3erHase) [inline test/unit/BPool.t.sol:732]: perhaps store factory in a `address` var?

File: test/unit/BPool.t.sol [hunk 3]
```diff
@@ -702,32 +702,134 @@ contract BPool_Unit_ExitPool is BasePoolTest {
...
+    bPool.exitPool(_fuzz.poolAmountIn, _zeroAmountsArray());
 
-  function test_Set_TokenArrayBalance() private view {}
+    assertEq(bPool.balanceOf(bPool.call__factory()), _balanceBefore - _fuzz.poolAmountIn + _exitFee);
+  }
 
-  function test_Emit_TokenArrayLogExit() private view {}
+  function test_Burn_PoolShare(ExitPool_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    uint256 _exitFee = bmul(_fuzz.poolAmountIn, EXIT_FEE);
+    uint256 _pAiAfterExitFee = bsub(_fuzz.poolAmountIn, _exitFee);
```
Comment (wei3erHase) [inline test/unit/BPool.t.sol:741]: what's `pAi` related to?

File: test/unit/BPool.t.sol [hunk 4]
```diff
@@ -702,32 +702,134 @@ contract BPool_Unit_ExitPool is BasePoolTest {
...
+    _setValues(_fuzz);
+
+    vm.expectRevert('ERR_MATH_APPROX');
+    bPool.exitPool(_fuzz.poolAmountIn, _zeroAmountsArray());
+  }
+
+  function test_Revert_TokenArrayLimitOut(ExitPool_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    vm.expectRevert('ERR_LIMIT_OUT');
+    bPool.exitPool(_fuzz.poolAmountIn, _maxAmountsArray());
+  }
```
Comment (wei3erHase) [inline test/unit/BPool.t.sol:765]: hmmm ok for now i'd say, but it'd be interesting to see that calculating the `balancesOut`, we can send it and won't revert, but if we ++ any value in the array, it'll revert

this expectation is too broad

**Review by wei3erHase — COMMENTED** [review]
looking good! small comments

File: foundry.toml [hunk 5]
```diff
@@ -27,8 +27,8 @@ out = 'out-via-ir'
 src = 'src/interfaces/'
 
 [fuzz]
-runs = 1000
-max_test_rejects = 1_000_000
+runs = 800
+max_test_rejects = 2_000_000
```
Comment (wei3erHase) [inline foundry.toml:31]: which test requires this?

File: test/unit/BPool.t.sol (hunk 3 above)
Comment (0xAustrian) [reply test/unit/BPool.t.sol:741]: it means poolAmountIn

File: foundry.toml (hunk 5 above)
Comment (0xAustrian) [reply foundry.toml:31]: The entire `BPool_Unit_ExitPool` contract, I tried keeping the runs at 1k but would have to increase the `max_rejects` a lot more

File: test/unit/BPool.t.sol (hunk 4 above)
Comment (0xAustrian) [reply test/unit/BPool.t.sol]: Check now

File: test/unit/BPool.t.sol [hunk 6]
```diff
@@ -759,9 +760,27 @@ contract BPool_Unit_ExitPool is BasePoolTest {
...
+      uint256 _bal = _fuzz.balance[i];
+      uint256 _tokenAmountOut = bmul(_ratio, _bal);
+
+      _minAmounts[i] = _tokenIndex == i ? _tokenAmountOut + 1 : _tokenAmountOut;
+    }
+
     vm.expectRevert('ERR_LIMIT_OUT');
-    bPool.exitPool(_fuzz.poolAmountIn, _maxAmountsArray());
+    bPool.exitPool(_fuzz.poolAmountIn, _minAmounts);
   }
```
Comment (wei3erHase) [inline test/unit/BPool.t.sol:784]: noisssss! 👏

**Review by wei3erHase — APPROVED** [review]
👍

**Pull request merged**